src/mscorlib/shared/System/Guid.cs

```diff
@@ -1251,41 +1251,37 @@ public string ToString(string format, IFormatProvider provider)
             if (format.Length != 1)
                 throw new FormatException(SR.Format_InvalidGuidFormatSpecification);
 
-            string guidString;
-
+            int guidSize;
             switch (format[0])
             {
                 case 'D':
                 case 'd':
-                    guidString = string.FastAllocateString(36);
+                    guidSize = 36;
                     break;
                 case 'N':
                 case 'n':
-                    guidString = string.FastAllocateString(32);
+                    guidSize = 32;
                     break;
                 case 'B':
                 case 'b':
                 case 'P':
                 case 'p':
-                    guidString = string.FastAllocateString(38);
+                    guidSize = 38;
                     break;
                 case 'X':
                 case 'x':
-                    guidString = string.FastAllocateString(68);
```

Comment:
It is better (smaller code) to call the method just once after the switch statement.